merlin-server/src/main/java/gov/nasa/jpl/aerie/merlin/server/remotes/postgres/PostProfileSegmentsAction.java

```diff
@@ -46,13 +47,32 @@ public <Dynamics> void apply(
       this.statement.setLong(2, profileRecord.id());
       setTimestamp(this.statement, 3, timestamp);
       setTimestamp(this.statement, 4, simulationStart);
-      this.statement.setString(5, serializeDynamics(dynamics, dynamicsP));
+      if (dynamics.isPresent()) {
+        this.statement.setString(5, serializeDynamics(dynamics.get(), dynamicsP));
+        this.statement.setBoolean(6, false);
+      } else {
+        this.statement.setString(5, "null");
+        this.statement.setBoolean(6, true);
+      }
 
       this.statement.addBatch();
 
       accumulatedOffset = Duration.add(accumulatedOffset, duration);
     }
 
+    // Profiles must explicitly end in a gap, because the last segment's duration data is lost when
+    // inserting into the database.
+    if (segments.get(segments.size()-1).getValue().isPresent()) {
```

Comment:
Theoretically none but it would take a bit more refactoring. Currently the `dataset` holds the start time, and its assumed that all profiles in the dataset start at the same time and end at infinity (but get truncated at the planning horizon). We could store a `duration` in `profile`? That seems a little cleaner than I have here. I only took this approach because it seemed like the smallest change that would do it.

Reply:
If it's not too much work, I do think it would be less surprising for the `duration` to be defined at the profile level, 

I've been waffling on this, wondering if your way would be more conducive to streaming, but I think the profile duration interpretation would be more consistent in either case (no profile segment would have special meaning as the "end" segment)

Reply:
More musings that might not be moving us towards a decision...

I suppose the interpretation is that the last segment applies until infinity. If your last segment is non-null, you are claiming that that segment applies until infinity. If your last segment is null, you are explicitly saying "I don't know the value from here until infinity". 

That interpretation matches your current implementation reasonably well - the part I'm not sure about is the requirement that the last segment be null...

Reply:
Yes, I'd also like to make sure that when we begin streaming results into the database from the simulation worker, we can easily append new segments to the profile with minimal overhead.

I'm a little nervous that including an explicit gap could require *replacing* that row when streaming new data into that area (e.g. if the current segment is merely extended rather than the profile taking on a new value). If I recall correctly, we don't partition the table of profile segments on a per-resource basis (that's far too fine-grained), only on a per-simulation basis. That means we might have to go hunting for the null row that needs to get updated; it's not guaranteed to be at the end.

For that reason, I prefer giving the `profile` field a duration -- acknowledging that this may also require more writes by the worker to subsequently update that duration. I think an append-only regime is more important.

Regardless of approach, I do think we need to delimit the interval of time over which we *actually* have results; if we haven't determined the value of a profile past time T, we ought not claim that the last segment extends beyond it (especially not to infinity). It *might*, but we don't know.

Reply:
Thanks for weighing in, @Twisol , that makes me feel better about our decision to go forward with the `duration` column on profile.

Reply:
done